astropy/utils/masked/core.py

```diff
@@ -567,16 +567,24 @@ def view(self, dtype=None, type=None):
             return super().view(self._get_masked_cls(type))
 
         dtype = np.dtype(dtype)
-        if not (
-            dtype.itemsize == self.dtype.itemsize
-            and (dtype.names is None or len(dtype.names) == len(self.dtype.names))
+        result = super().view(dtype, self._get_masked_cls(type))
+        # Mask should be viewed in all but simplest case.
+        if (
+            dtype.itemsize != self.dtype.itemsize
+            or dtype.names
+            or dtype.shape
+            or self.dtype.names
+            or self.dtype.shape
         ):
-            raise NotImplementedError(
-                f"{self.__class__} cannot be viewed with a dtype with a "
-                "with a different number of fields or size."
-            )
+            try:
+                result.mask = self.mask.view(np.ma.make_mask_descr(dtype))
+            except Exception as exc:
+                raise NotImplementedError(
+                    f"{self.__class__} cannot be viewed with a dtype with a "
+                    "with a different number of fields or size."
+                ) from exc
```

Comment:
I think this chaining doesn't make the message clearer, but feel free to ignore if you feel strongly about it.
```suggestion
                ) from None
```

Reply:
Yes, agreed, the viewing error message will not really help.